### PR TITLE
phpDocumentor

### PR DESCRIPTION
**Pull request by waltertamboer**
This is going to be an enormous PR so maybe you guys can keep reviewing.

@psren @pachico

### CODE REVIEW

File: phpdoc.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <files>
+        <directory>src/</directory>
+    </files>
+    <transformer>
+        <target>docs/api/</target>
+    </transformer>
+    <transformations>
+        <template name="responsive-twig"/>
+    </transformations>
+</phpdoc>
```
Comment (HellPat) [inline phpdoc.xml]: newline at end of file?

File: src/Analytics/DB/DataCollector.php [hunk 2]
```diff
@@ -15,7 +27,7 @@ class DataCollector implements SubscriberInterface
     private $participations = [];
 
     /**
-     * @inheritDoc
+     * {@inheritDoc}
```
Comment (HellPat) [inline src/Analytics/DB/DataCollector.php:30]: Hah, i never understood the difference.

File: src/Analytics/DB/DataCollector.php [hunk 3]
```diff
@@ -35,22 +47,23 @@ public function getSubscribedEvents()
     }
 
     /**
+     * Adds a participation to the data collector.
+     *
      * @param string $testIdentifier
      * @param string $variationIdentifier
      * @throws InvalidArgumentException
-     *
-     * @return void
```
Comment (HellPat) [inline src/Analytics/DB/DataCollector.php:42]: was this wrong?

File: src/Participation/FilterInterface.php [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
+ *
+ * @package PhpAb
  */
 interface FilterInterface
 {
     /**
      * Checks if a user should participate in the test
      *
-     * @return boolean
+     * @return boolean Returns true when the user should participate; false otherwise.
```
Comment (HellPat) [inline src/Participation/FilterInterface.php:22]: "Returns" if obvious here

Comment (HellPat) [inline src/Participation/FilterInterface.php:22]: Okay, reconsidered. Returns is totally okay

**Comment by HellPat** [comment]
Super cool so far :-)

File: phpdoc.xml (hunk 1 above)
Comment (waltertamboer) [inline phpdoc.xml]: I've added the new line, weird that your comment does not disappear.

File: src/Analytics/DB/DataCollector.php (hunk 2 above)
Comment (waltertamboer) [inline src/Analytics/DB/DataCollector.php:30]: Yeah, the one does not work and the other does :)

File: src/Analytics/DB/DataCollector.php (hunk 3 above)
Comment (waltertamboer) [inline src/Analytics/DB/DataCollector.php:42]: Not per se. It's just not needed. When no return type is provided, the default return type is void.

**Comment by waltertamboer** [comment]
I've removed the WIP flag.

**Pull request merged**